tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/TypedPredefs.scala

```diff
@@ -0,0 +1,145 @@
+package at.forsyte.apalache.tla.typecheck
+
+import at.forsyte.apalache.tla.lir.values._
+import at.forsyte.apalache.tla.lir._
+
+object TypedPredefs {
+  type Tag = Typed[TlaType1]
+
+  /**
+   * An implicit conversion of an expression to a builder block
+   *
+   * @param ex a TLA+ expression
+   * @return a builder block
+   */
+  implicit def tlaExToBlock(ex: TlaEx): BuilderTlaExWrapper = {
+    BuilderTlaExWrapper(ex)
+  }
+
+  /**
+   * An implicit conversion of a sequence of expressions to a sequence of builder blocks.
+   *
+   * @param es a sequence of TLA+ expressions
+   * @return the corresponding sequence of builder blocks
+   */
+  implicit def seqOfTlaExToSeqOfBlocks(es: Seq[TlaEx]): Seq[BuilderTlaExWrapper] = {
+    es.map(tlaExToBlock)
+  }
+
+  /**
+   * An implicit wrapper around TypeTag that extract the type out of Typed(_: TlaType1).
+   *
+   * @param tag a type tag
+   */
+  implicit class TypeTagAsTlaType1(tag: TypeTag) {
```

Comment:
`TypeTag`, `TlaEx`, etc. are declared in the IR that does not know about `TlaType1`. I prefer to keep it like that because we may want to introduce another type system in the future. Moreover, the model checker itself is using another (much simpler) type system. So it is good to separate types from expressions, at least for now. The parser should not know about types either, as it is completely untyped.